### PR TITLE
RN v61 upgrade

### PR DESCRIPTION
**Pull request by chrisbobbe**
~~Awaiting https://github.com/zulip/zulip-mobile/pull/4150.~~ And, it's merged! Thanks, @gnprice !

Fixes: #3781

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Hmm, actually, there's a part I haven't investigated yet—the various changes to `project.pbxproj`. I'll do that now.

**Comment by chrisbobbe** [comment]
Bookmarking this, as something we might (?) have to think about, for mocking native modules now that Haste isn't being used: https://altany.github.io/react-native/0.61/jest/mocking/upgrade/2020/01/25/mocking-react-native-0.61-modules-with-jest.html.

It links to https://github.com/facebook/react-native/issues/26579#issuecomment-535244001.

**Comment by chrisbobbe** [comment]
> **edit**: One way of doing this is at [e956f99](https://github.com/zulip/zulip-mobile/commit/e956f9912e9b5e6980fb25c501bbb8a9cfb28001) from my current revision of #4183 (not merged yet).

OK, I've cherry-picked [e956f99](https://github.com/zulip/zulip-mobile/commit/e956f9912e9b5e6980fb25c501bbb8a9cfb28001) and made some changes.

Also, I resolved some conflicts; this is ready for a review. 🙂

**Comment by gnprice** [comment]
Thanks @chrisbobbe ! Chat discussion starting [here](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M3781.20RN.20v0.2E61.20upgrade/near/931194). Looks good, with some comments in that thread:
* the changes to Jest mocking (details there)
* probably worth including the equivalent of facebook/react-native@9127fb51fc94a94c4a9f957c3e421a1667a9a20a
* a couple of items to mention in your list of changes we're omitting, at https://github.com/zulip/zulip-mobile/issues/3781#issuecomment-642151537

File: jest/jestSetup.js [hunk 1]
```diff
@@ -1,19 +1,67 @@
 import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';
+import * as ReactNative from 'react-native';
+
+// Following upstream advice [1], mock `react-native` ourselves.
+//
+// Helpful in particular for adding to NativeModules, to simulate
+// modules being available after the bridge initializes.
+//
+// The 'react-native' we import here is already mocked to a great
+// extent by `node_modules/react-native-jest/setup.js`, thanks to our
```
Comment (gnprice) [inline jest/jestSetup.js:10]:
```suggestion
// extent by `node_modules/react-native/jest/setup.js`, thanks to our
```

**Comment by chrisbobbe** [comment]
Thanks for the review! Just pushed these changes to this branch; also propagated them to my branch rn-61-followup-1, for #4152.

**Comment by chrisbobbe** [comment]
Thanks, @gnprice! I'll address the conflicts over at #4152, which will fix an existing issue!

**Pull request merged**